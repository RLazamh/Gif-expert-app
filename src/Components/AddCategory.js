import React, { useState } from 'react'
import PropTypes from 'prop-types'
function AddCategory({setCategories}) {
    // el ultimo valor actualizado del primero 
    // siempre sera el ultimo ingresado
    const [inputText, setInputText] = useState('');

    const handleInputText = (event) =>{
        setInputText(event.target.value);
    }
    const handleForm = (event)=>{
        event.preventDefault();

        if(inputText.trim().length > 2){
            setCategories((category)=>[inputText,...category]);
        }
    }
    return (
        <form onSubmit={handleForm}>
            <p>{inputText}</p>
          <input
                type = 'text'
                value={inputText}
                onChange ={handleInputText}
          />  
        </form>
    )
}

AddCategory.propTypes = {
    setCategories:PropTypes.func.isRequired

}

export default AddCategory

