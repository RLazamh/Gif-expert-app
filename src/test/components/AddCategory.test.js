import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import AddCategory from "../../Components/AddCategory";

describe('test in <AddCategory /> component', () => {

    const setCategories = jest.fn(); // es una funcion cuantas veces es llamada etc 
    let wrapper =  shallow (<AddCategory setCategories={setCategories}/>);
    beforeEach(()=>{
        jest.clearAllMocks
        wrapper =  shallow (<AddCategory setCategories={setCategories}/>);

    })
    test('should to match with snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should return value on Change',()=>{
        const input =  wrapper.find('input');
        const value = 'hola mundo';
        input.simulate('change',{ target: { value } });
        expect(wrapper.find('p').text().trim()).toBe(value)
    })
    test('No debe llamarse las funcion ', () => {
        wrapper.find('form').simulate('submit',{ preventDefault(){} })
        expect(setCategories).not.toHaveBeenCalled()
    })
    
})
