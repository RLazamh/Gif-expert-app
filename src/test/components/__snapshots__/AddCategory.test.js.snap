// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <AddCategory /> component should to match with snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
